lib/base_service.js

```diff
@@ -32,13 +33,38 @@ class BaseService {
         this._logger = null;
         this._metrics = null;
         this._ratelimiter = null;
-        this.serviceReturns = null;
+        this._serviceReturns = null;
+
+        // Avoid recursion if there are bugs in the logging code.
+        this._inLogger = false;
+
+        this._uncaughtExceptionHandler = (err) => {
```

Comment:
Why are we declaring this method inside the constructor? Let's be consistent and declare it just like all other methods.

Reply:
No. That's actually correct. If we declare the method like any other method, it will be a property of the object prototype thus shared between all instances of the master/worker. So we wouldn't be able to create several service-runners within the same process and add/remove appropriate event handlers on start/stop.